nbconvert/preprocessors/execute.py

```diff
@@ -440,6 +440,7 @@ def run_cell(self, cell, cell_index=0):
         exec_reply = self._wait_for_reply(msg_id, cell)
 
         outs = cell.outputs = []
+        self.clear_before_next_output = False
```

Comment:
Should we check if this is True before exiting (if it were the final message) and clear just before?

Reply:
No, if nothing is outputted, it should not clear.

Reply:
This should be covered by the notebook btw.

Reply:
I more meant if the final line of the final cell were `clear_output(wait=True)` wouldn't this not apply clear before exiting `run_cell`? I might be missing a behavioral aspect this morning before my coffee kicks in :D

Reply:
Yes, it would not clear, and it should not. One of the cells in the
notebook has this. Unless you found a scenario I did not test. (apparently, reply by email does not work well with threads).

Reply:
Ok thanks, looks good to me them. I'll get to looking at your other PRs related to widgets this next week (been super busy)